component/loki/source/journal/internal/target/journaltarget.go

```diff
@@ -0,0 +1,417 @@
+//go:build linux && cgo && promtail_journal_enabled
+// +build linux,cgo,promtail_journal_enabled
+
+package target
+
+// This code is copied from Promtail with minor edits. The target package is used to
+// configure and run the targets that can read journal entries and forward them
+// to other loki components.
```

Comment:
IMHO it might a good idea to also port the test code from Promtail. This would offer us slightly more confidence in making changes in the future when bringing in new fixes. WDYT?

https://github.com/grafana/loki/blob/main/clients/pkg/promtail/targets/journal/journaltarget_test.go

Reply:
Sounds good